lib/WiFi/src/WiFiEventHandler.cpp

```diff
@@ -0,0 +1,268 @@
+    if (ctx == nullptr) {
+        ESP_LOGD(LOG_TAG, "No context");
+        return ESP_OK;
+    }
+
+    esp_err_t rc = ESP_OK;
+    switch (event->event_id) {
+    case SYSTEM_EVENT_AP_START: {
+        rc = p_wifi_event_handler->apStart();
+        break;
+    }
+
+    case SYSTEM_EVENT_AP_STOP: {
+        rc = p_wifi_event_handler->apStop();
+        break;
+    }
+
+    case SYSTEM_EVENT_AP_STACONNECTED: {
+        rc = p_wifi_event_handler->apStaConnected(event->event_info.sta_connected);
+        break;
+    }
+
+    case SYSTEM_EVENT_AP_STADISCONNECTED: {
+        rc = p_wifi_event_handler->apStaDisconnected(event->event_info.sta_disconnected);
+        break;
+    }
+
+    case SYSTEM_EVENT_SCAN_DONE: {
+        rc = p_wifi_event_handler->staScanDone(event->event_info.scan_done);
+        break;
+    }
+
+    case SYSTEM_EVENT_STA_AUTHMODE_CHANGE: {
+        rc = p_wifi_event_handler->staAuthChange(event->event_info.auth_change);
+        break;
+    }
+
+    case SYSTEM_EVENT_STA_CONNECTED: {
+        rc = p_wifi_event_handler->staConnected(event->event_info.connected);
+        break;
+    }
+
+    case SYSTEM_EVENT_STA_DISCONNECTED: {
+        rc = p_wifi_event_handler->staDisconnected(event->event_info.disconnected);
+        break;
+    }
+
+    case SYSTEM_EVENT_STA_GOT_IP: {
+        rc = p_wifi_event_handler->staGotIp(event->event_info.got_ip);
+        break;
+    }
+
+    case SYSTEM_EVENT_STA_START: {
+        rc = p_wifi_event_handler->staStart();
+        break;
+    }
+
+    case SYSTEM_EVENT_STA_STOP: {
+        rc = p_wifi_event_handler->staStop();
+        break;
+    }
+
+    case SYSTEM_EVENT_WIFI_READY: {
+        rc = p_wifi_event_handler->wifiReady();
+        break;
+    }
+
+    default:
+        break;
+    }
+
+    if (p_wifi_event_handler->m_next_handler != nullptr) {
+        ESP_LOGD(LOG_TAG, "Found a next handler");
+        rc = eventHandler(p_wifi_event_handler->m_next_handler, event);
+    } else {
+        //ESP_LOGD(LOG_TAG, "NOT Found a next handler");
+    }
+    return rc;
+} // eventHandler
+
+
+/**
+ * @brief Constructor
+ */
+WiFiEventHandler::WiFiEventHandler()
+{
+    m_next_handler = nullptr;
+} // WiFiEventHandler
+
+
+/**
+ * @brief Get the event handler.
+ * Retrieve the event handler function to be passed to the ESP-IDF event handler system.
+ * @return The event handler function.
+ */
+system_event_cb_t WiFiEventHandler::getEventHandler()
+{
+    ESP_LOGD(LOG_TAG, "<< getEventHandler: 0x%x", (uint32_t)eventHandler);
+    return eventHandler;
+} // getEventHandler
+
+
+/**
+ * @brief Handle the Station Got IP event.
+ * Handle having received/assigned an IP address when we are a station.
+ * @param [in] event_sta_got_ip The Station Got IP event.
+ * @return An indication of whether or not we processed the event successfully.
+ */
+esp_err_t WiFiEventHandler::staGotIp(system_event_sta_got_ip_t info)
+{
+    ESP_LOGD(LOG_TAG, "default staGotIp");
+    return ESP_OK;
+} // staGotIp
+
+
+/**
```

Comment:
Move this and other comments here in the `WiFiEventHandler.cpp` file to `WiFiEventHandler.h`, so the documentation gets generated